src/detection/FootprintMerge.cc

```diff
@@ -247,8 +247,8 @@ void FootprintMergeList::addCatalog(std::shared_ptr<afw::table::SourceTable> sou
                           (boost::format("Filter %s not in original list") % filter).str());
     }
 
-    // If list is empty don't check for any matches, just add all the objects
-    bool checkForMatches = (_mergeList.size() > 0);
+    // If list is empty or merging not requested, don't check for any matches, just add all the objects
+    bool checkForMatches = (_mergeList.size() > 0) && doMerge;
```

Comment:
Prefer `!_mergeList.empty()`; this comparison will generate a signed vs. unsigned warning on some compilers.